### PR TITLE
list unsafe strings as xml variables

### PR DESCRIPTION
**Pull request by sravfeyn**
http://manage.dimagi.com/default.asp?pre=preMultiSearch&pg=pgList&pgBack=pgSearch&search=2&searchFor=237863#1272942

@calellowitz 
@snopoke can you look at this as well.

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_suite.py [hunk 1]
```diff
@@ -339,6 +339,62 @@ def test_case_detail_tabs_with_nodesets(self):
...
+                ],
+                case_tile_field='sex'
+            ),
+        ]
+
+        spec = """
+            <partial>
+              <template>
+                <text>
+                  <xpath function="if(color = 'red', $kred, if(color = 'with space', $h{hash_val2}, if(color = $e{hash_val3}, $h{hash_val3}, '')))">
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_suite.py]: E501 line too long (148 > 115 characters)

Comment (sravfeyn) [inline corehq/apps/app_manager/tests/test_suite.py]: (Ignoring)

**Review by snopoke — COMMENTED** [review]
LGTM though I was wondering if it isn't worth trying combine this with `MappingItem.key_as_condition` which gets used by `EnumImage`.

**Comment by snopoke** [comment]
Real build failure:
```
======================================================================
FAIL: test_case_detail_icon_mapping (corehq.apps.app_manager.tests.test_suite.SuiteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_suite.py", line 449, in test_case_detail_icon_mapping
    './detail/field/template[@form="image"]'
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 26, in assertXmlPartialEqual
    self.assertXmlEqual(expected, actual, normalize=False)
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 32, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 126, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -11,6 +11,7 @@
         <variable name="k10">
           <locale id="m0.case_short.case_age_1.enum.k10"/>
         </variable>
+        <variable name="e62796255">age &amp;gt; 50</variable>
       </xpath>
     </text>
   </template>
```

**Comment by sravfeyn** [comment]
The reason for not using `key_as_condition` similar to `EnumImage` is because, in terms of feature spec, advanced calculations are to be allowed in Calculated Icons but not in ID-Mapping (though I don't know why that is). There is also this new feature request http://manage.dimagi.com/default.asp?245496#1273477. I am closing this for now, so that I can consider these two together when I work on it.

The build failure needs a test-case update, it looks like

**Pull request closed without merging**